### PR TITLE
(330) Health task: substance misuse page

### PR DESCRIPTION
**Pull request by edavey**
This PR implements the first real form page in the "Health needs" task: "Substance misuse". (The initial page is a guidance page, and is not included in the left hand navigation).

See [Trello 330](https://trello.com/c/Qg994Nnx/330-health-subsection-substance-misuse)

substance_misuse

### CODE REVIEW

**Comment by edavey** [comment]
[Health subsection - Substance misuse](https://trello.com/c/Qg994Nnx/330-health-subsection-substance-misuse)

File: server/form-pages/apply/risks-and-needs/health-needs/substanceMisuse.test.ts [hunk 1]
```diff
@@ -13,6 +13,40 @@ describe('SubstanceMisuse', () => {
     })
   })
 
+  describe('Questions', () => {
+    const page = new SubstanceMisuse({}, application)
+
+    describe('usesIllegalSubstances', () => {
+      it('has a question', () => {
+        expect(page.questions.usesIllegalSubstances.question).toBeDefined()
```
Comment (patrickjfl) [inline server/form-pages/apply/risks-and-needs/health-needs/substanceMisuse.test.ts:21]: I'm afraid I instinctively don't really like the look of toBeDefined(). What do you think of exporting the questions variable from the substanceMisuse file, importing it into here and checking the values match?

File: integration_tests/pages/apply/risks-and-needs/substanceMisusePage.ts [hunk 2]
```diff
@@ -19,4 +19,20 @@ export default class SubstanceMisusePage extends ApplyPage {
...
     )
   }
+
+  describeIllegalSubstanceUse = (): void => {
+    this.checkRadioByNameAndValue('usesIllegalSubstances', 'yes')
+    this.getTextInputByIdAndEnterDetails('substanceMisuseHistory', 'Heroin user')
+    this.getTextInputByIdAndEnterDetails('substanceMisuseDetail', 'Injects daily')
+  }
+
+  nameDrugAndAlcholService = (): void => {
```
Comment (patrickjfl) [inline integration_tests/pages/apply/risks-and-needs/substanceMisusePage.ts:29]: typo in alcohol :P

**Review by patrickjfl — APPROVED** [review]
Very nice and clear, well broken up. Just a few minor comments. Also, your final commit comment is really helpful documentation - could we add it to the existing doc on apply / form pages?

File: server/form-pages/apply/risks-and-needs/health-needs/substanceMisuse.test.ts (hunk 1 above)
Comment (edavey) [reply server/form-pages/apply/risks-and-needs/health-needs/substanceMisuse.test.ts:21]: hmm, interesting point, but I'm not sure that asserting that:

```
expect(
  page.
    questions.usesIllegalSubstances.question).toEqual(
    importedQuestions.usesIllegalSubstances.question
  )
```

is any more meaningful?

My intention with this spec is to describe the structure of the questions: that there's one question asking whether illegal substances are being used, and within that 2 followup questions etc. It's deliberately not concerned with the type of question or the wording of the question as those are going to be covered in the description of the `response()`. 

If we document the precise wording of the questions in two places we will increase the maintenance burden -- with no benefit which I can see.

But maybe I am misunderstanding?

File: server/form-pages/apply/risks-and-needs/health-needs/substanceMisuse.ts [hunk 3]
```diff
@@ -1,17 +1,58 @@
...
+    'requiresSubstituteMedication',
+    'substituteMedicationDetail',
+  ],
 })
 export default class SubstanceMisuse implements TaskListPage {
   title = `Health needs for ${this.application.person.name}`
 
+  questions = {
+    usesIllegalSubstances: {
+      question: 'Do they take any illegal substances?',
```
[the remark this answers is not in the document]
Comment (edavey) [reply server/form-pages/apply/risks-and-needs/health-needs/substanceMisuse.ts:34]: Another good (difficult) question! What I've done is to:

- go by the prototype for now for all for all the question text
- come up with error messages myself
- add a personalised heading to the page ("Health needs for Roger Smith")

In the interest of moving forward to get a first cut of the whole "Health needs" task done I'd l prefer to put this in front of the team and then come back around again as necessary with a round of changes.

File: integration_tests/pages/apply/risks-and-needs/substanceMisusePage.ts [hunk 4]
```diff
@@ -19,4 +19,20 @@ export default class SubstanceMisusePage extends ApplyPage {
...
+    this.checkRadioByNameAndValue('engagedWithDrugAndAlcoholService', 'yes')
+    this.getTextInputByIdAndEnterDetails('drugAndAlcoholServiceDetail', 'The Drugs Project')
+  }
+
+  provideSubsituteMedicationDetails = (): void => {
```
[the remark this answers is not in the document]
Comment (edavey) [reply integration_tests/pages/apply/risks-and-needs/substanceMisusePage.ts:34]: thank you!

File: integration_tests/pages/apply/risks-and-needs/substanceMisusePage.ts (hunk 2 above)
Comment (edavey) [reply integration_tests/pages/apply/risks-and-needs/substanceMisusePage.ts:29]: nicely spotted!

File: server/form-pages/apply/risks-and-needs/health-needs/substanceMisuse.test.ts (hunk 1 above)
Comment (patrickjfl) [reply server/form-pages/apply/risks-and-needs/health-needs/substanceMisuse.test.ts:21]: ah yes, of course that's covered in the response() test. OK, I see what you're saying and I'm happy with that

File: server/form-pages/apply/risks-and-needs/health-needs/substanceMisuse.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (patrickjfl) [reply server/form-pages/apply/risks-and-needs/health-needs/substanceMisuse.ts:34]: ok sure, makes sense 👍

**Pull request merged**